pkg/deps/config/suppliers.go

```diff
@@ -138,17 +138,17 @@ func NewSupplyQueryClientContextFn(pocketQueryNodeURL string) SupplierFn {
 // NewSupplyTxClientContextFn returns a function with constructs a ClientContext
 // instance with the given cmd and returns a new depinject.Config which is
 // supplied with the given deps and the new ClientContext.
-func NewSupplyTxClientContextFn(pocketTxNodeURL string) SupplierFn {
+func NewSupplyTxClientContextFn(txNodeGRPCURL *url.URL) SupplierFn {
 	return func(_ context.Context,
 		deps depinject.Config,
 		cmd *cobra.Command,
 	) (depinject.Config, error) {
 		// Temporarily store the flag's current value
-		tmp := cosmosflags.FlagNode
+		tmpGRPC := cosmosflags.FlagGRPC
 
 		// Set --node flag to the pocketTxNodeURL for the client context
 		// This flag is read by cosmosclient.GetClientTxContext.
-		if err := cmd.Flags().Set(cosmosflags.FlagNode, pocketTxNodeURL); err != nil {
+		if err := cmd.Flags().Set(cosmosflags.FlagGRPC, txNodeGRPCURL.Host); err != nil {
```

Comment:
Would you mind adding a comment here that makes it clear that we're intentionally only using the host because cosmos-sdk expects the format `<hostname>[:<port>]`.